src/io/flutter/logging/FlutterErrorHelper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.logging;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.flutter.inspector.DiagnosticLevel;
+import io.flutter.inspector.DiagnosticsNode;
+
+import java.util.regex.Pattern;
+
+public class FlutterErrorHelper {
+  private static final Pattern numberPattern = Pattern.compile("[0-9]+");
+
+  public static String getAnalyticsId(DiagnosticsNode node) {
+    for (DiagnosticsNode property : node.getInlineProperties()) {
+      if (property.getLevel() == DiagnosticLevel.summary) {
+        final String description = property.getDescription();
+        return getAnalyticsId(description);
+      }
+    }
+
+    return null;
+  }
+
+  @VisibleForTesting
+  public static String getAnalyticsId(String errorSummary) {
+    // "A RenderFlex overflowed by 1183 pixels on the right."
+
+    // normalize to lower case
```

Comment:
@InMatrix @jacob314 please review the normalization we're doing below

Reply:
Lets add a TODO to remove the normalization. With a future change adding a kernel transformer to Flutter for the error description objects, this error summary will start showing up as
`'A $runtimeType overflowed by $overflowText.'`

Reply:
👍 done